sw/device/lib/dif/dif_csrng.h

```diff
@@ -0,0 +1,550 @@
+
+/**
+ * Creates a new handle for CSRNG.
+ *
+ * This function does not actuate the hardware.
+ *
+ * @param params Hardware instantiation parameters.
+ * @param[out] csrng Out param for the initialized handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_csrng_result_t dif_csrng_init(dif_csrng_params_t params,
+                                  dif_csrng_t *csrng);
+
+/**
+ * Configures CSRNG with runtime information.
+ *
+ * This function should need to be called once for the lifetime of `handle`.
+ *
+ * @param csrng A CSRNG handle.
+ * @param config Runtime configuration parameters.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_csrng_result_t dif_csrng_configure(const dif_csrng_t *csrng,
+                                       dif_csrng_config_t config);
+
+/**
+ * Initializes CSRNG instance with a new seed value.
+ *
+ * `params.seed_material` is used as specified in NIST SP 800-90Ar1 section
+ * 10.2.1.3.1. See `dif_csrng_instantiate_source` for details on how this value
+ * is mixed with the CSRNG seed.
+ *
+ * @param csrng A CSRNG handle
+ * @param params Seed initialization parameters.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_csrng_result_t dif_csrng_instantiate(
+    const dif_csrng_t *csrng, const dif_csrng_instantiate_source_t *seed_source,
+    const dif_csrng_transaction_t *params);
+
+/**
+ * Reseeds CSRNG instance.
+ *
+ * When `params.seed_material_len` is set to 0, only the entropy source
+ * seed is used to reseed the instance.
+ *
+ * @param csrng A CSRNG handle.
+ * @param params Reseed parameters.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_csrng_result_t dif_csrng_reseed(const dif_csrng_t *csrng,
+                                    const dif_csrng_transaction_t *params);
+
+/**
+ * Updates CSRNG state.
+ *
+ * This function is similar to `dif_csrng_reseed()`, except:
+ *
+ * - Only `params.seed_material` is used in the update operation.
+ * - The update operation does not reset the internal CSRNG reseed
+ *   counter.
+ *
+ * @param csrng A CSRNG handle.
+ * @param params Update parameters.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_csrng_result_t dif_csrng_update(const dif_csrng_t *csrng,
+                                    const dif_csrng_transaction_t *params);
+
+/**
+ * Requests cryptographic entropy bits from the CSRNG.
+ *
+ * The prediction resistance flag as specified in SP 800-90Ar1 section
+ * 10.2.1.1 is not directily supported by the hardware. It is the
+ * responsibility of the caller to reseed as needed before calling
+ * this function.
+ *
+ * The CSRNG accepts gereration requets with 128-bit granularity, with
+ * a minimum 128-bit request size. This function will increase the size
+ * of the request to align it to the nearest 128-bit boundary.
+ *
+ * @param csrng A CSRNG handle.
+ * @param len Number of bytes to generate.
```

Comment:
Here `len` refers to bytes, whereas for `dif_csrng_read_output`, `len` refers to words.

Does it make sense to regularize the meaning of `len` between these two apis?

Reply:
SGTM.  Updated to number of `uint32_t` words for consistency.